L1Trigger/TrackerTFP/interface/DataFormats.h

```diff
@@ -0,0 +1,688 @@
+    // sectors this stub belongs to
+    std::vector<int> sectors() const { return sectors_.ids(); }
+    // stub radius wrt chosenRofPhi
+    double r() const { return std::get<0>(data_); }
+    // stub phi wrt processing nonant centre
+    double phi() const { return std::get<1>(data_); }
+    // stub z
+    double z() const { return std::get<2>(data_); }
+    // reduced layer id
+    int layer() const { return std::get<3>(data_); }
+    // phi sector map to which this stub belongs to
+    TTBV sectorsPhi() const { return std::get<4>(data_); }
+    // first eta sector this stub belongs to
+    int sectorEtaMin() const { return std::get<5>(data_); }
+    // last eta sector this stub belongs to
+    int sectorEtaMax() const { return std::get<6>(data_); }
+    // first inv2R bin this stub belongs to
+    int inv2RMin() const { return std::get<7>(data_); }
+    // last inv2R bin this stub belongs to
+    int inv2RMax() const { return std::get<8>(data_); }
+  private:
+    // sectors this stub belongs to
+    TTBV sectors_;
+  };
+
+  // class to represent stubs generated by process geometric processor
+  class StubGP : public Stub<double, double, double, int, int, int> {
+  public:
+    // construct StubGP from Frame
+    StubGP(const TTDTC::Frame& frame, const DataFormats* dataFormats, int sectorPhi, int sectorEta);
+    // construct StubGO from StubPP
+    StubGP(const StubPP& stub, int sectorPhi, int sectorEta);
+    ~StubGP(){}
+    // true if stub belongs to given inv2R bin
+    bool inInv2RBin(int inv2RBin) const { return inv2RBins_[inv2RBin]; }
+    // inv2R bins  this stub belongs to
+    std::vector<int> inv2RBins() const { return inv2RBins_.ids(); }
+    // stub phi sector
+    int sectorPhi() const { return sectorPhi_; }
+    // stub eta sector
+    int sectorEta() const { return sectorEta_; }
+    // stub radius wrt chosenRofPhi
+    double r() const { return std::get<0>(data_); }
+    // stub phi wrt phi sector centre
+    double phi() const { return std::get<1>(data_); }
+    // stub z residual wrt eta sector
+    double z() const { return std::get<2>(data_); }
+    // reduced layer id
+    int layer() const { return std::get<3>(data_); }
+    // first inv2R bin this stub belongs to
+    int inv2RMin() const { return std::get<4>(data_); }
+    // last inv2R bin this stub belongs to
+    int inv2RMax() const { return std::get<5>(data_); }
+  private:
+    // inv2R bins this stub belongs to
+    TTBV inv2RBins_;
+    // stub phi sector
+    int sectorPhi_;
+    // stub eta sector
+    int sectorEta_;
+  };
+
+  // class to represent stubs generated by process hough transform
+  class StubHT : public Stub<double, double, double, int, int, int, int> {
+  public:
+    // construct StubHT from Frame
+    StubHT(const TTDTC::Frame& frame, const DataFormats* dataFormats, int inv2R);
+    // construct StubHT from StubGP
+    StubHT(const StubGP& stub, int inv2R, int phiT);
+    ~StubHT(){}
+    // stub qOver pt
```

Comment:
Suggest moving inv2R() to after declaration of phiT(), as is already done in class StubMHT. This groups together the two "track" variables related to the HT cell the stub is assigned to. Also add comment above them. "HT cell assignment"

Reply:
done

Reply:
This seems not to have been done? Now at L400.

Reply:
since inv2R is not part of the h/w stub while phiT is it comes after to be consistent with other cases where we have an stub / track with additional data which is not part of the h/w word.